### PR TITLE
Initial server-side session ticket support

### PR DESCRIPTION
**Pull request by cpu**
This branch starts to sketch out initial server-side support for TLS session tickets. We take the easy route for now and manage this opaquely internal to Rustls using the aws-lc-rs implementation of an AEAD ticketer/ticket switcher. 

We offer the ability for OpenSSL API users to disable ticket support (either with `SSL_set_options()` and `SSL_OP_NO_TICKET`, or with `SSL_CONF_cmd` using `-no_tickets` or `Options -SessionTicket`) and to control the number of session tickets (defaulting to 2 to match OpenSSL default, customizable with `SSL_CTX_set_num_tickets`/`SSL_set_num_tickets`). 

We don't offer more fine grain control over generating/encrypting/decrypting tickets and instead stub `SSL_CTX_set_tlsext_ticket_key_cb` and `SSL_CTX_set_tlsext_ticket_key_evp_cb`.

Resolves https://github.com/rustls/rustls-openssl-compat/issues/30

### CODE REVIEW

**Comment by cpu** [comment]
> The [miri task is unhappy](https://github.com/rustls/rustls-openssl-compat/actions/runs/9687156593/job/26731147955?pr=35) with the invocation of the aws_lc_rs ticketer constructor: "can't call foreign function aws_lc_0_15_0_RAND_bytes on OS linux"

I thought I might be able to add a `[no_mangle]` stub for this in `miri.rs` but haven't had luck. I notice with closer examination the CI failure output indicates the symbol as `aws_lc_0_15_0_RAND_bytes` - that first bit of sort-of-mojibake-looking-crap seems to be a [U+0001](https://www.compart.com/en/unicode/U+0001). I'm not sure why that's there, or if it has any implication for why I can't export a matching symbol name. In any case I worked around this for now by just not initializing a ticketer for the case where `cfg!(miri)` is set.

**Comment by cpu** [comment]
> I'm either misunderstanding the SSL_CONF_cmd(cctx, "Options", "SessionTicket") semantics

This appears to be the case. I was only setting `SSL_CONF_FLAG_FILE` on the `SSL_CONF_CTX` but the 2nd level logic that's handling the `SessionTicket` options flag value in particular seems smart enough to know it requires `SSL_CONF_FLAG_SERVER` and returns 0 otherwise - I'll make some adjustments.

**Comment by cpu** [comment]
I think this is ready for review now.

File: rustls-libssl/tests/config.c [hunk 1]
```diff
@@ -14,38 +14,46 @@ static const int conf_flags[] = {SSL_CONF_FLAG_SERVER, SSL_CONF_FLAG_CLIENT,
...
-                                                     "-verifyCApath",
-                                       CUSTOM_PREFIX "verifyCApath",
-                                       "VerifyCAPath",
-                                       CUSTOM_PREFIX "VerifyCAPath",
-
-                                       "-verifyCAfile",
-                                       CUSTOM_PREFIX "verifyCAfile",
-                                       "VerifyCAFile",
-                                       CUSTOM_PREFIX "VerifyCAFile"};
+static const char *supported_cmds[] = {
```
Comment (ctz) [inline rustls-libssl/tests/config.c:18]: ahah, it's back being sensible. is it the length that controls this? or the trailing comma?

**Review by ctz — APPROVED** [review]

File: rustls-libssl/tests/config.c (hunk 1 above)
Comment (cpu) [reply rustls-libssl/tests/config.c:18]: > is it the length that controls this? or the trailing comma?

I don't think it's the trailing comma. I've been unable to determine the formatting logic (or how to influence it). The results are UGLY in all instances :sob:

**Pull request merged**